kolibri/plugins/setup_wizard/assets/src/views/ImportMultipleUsers.vue

```diff
@@ -0,0 +1,208 @@
+      circular loader; then when they are done being imported, we add them to the final state
+      which allows us to identify them as being "imported" in place of the "import" button
+    - If the admin goes back from here they go to a loading page which will ping the same Queue
+      and offer them to import another user once all SoudQueue tasks are COMPLETE
+  */
+  export default {
+    name: 'ImportMultipleUsers',
+    components: {
+      OnboardingStepBase,
+      PaginatedListContainer,
+      UserTable,
+    },
+    mixins: [commonCoreStrings, commonSyncElements],
+
+    data() {
+      return {
+        isPolling: false,
+        // array of user/learner ids
+        learnersBeingImported: [],
+      };
+    },
+    inject: ['wizardService'],
+    computed: {
+      learners() {
+        return this.wizardService.state.context.remoteUsers;
+      },
+      usersID() {
+        return this.learners.map(user => user.id);
+      },
+      device() {
+        return this.wizardService.state.context.importDevice;
+      },
+      facility() {
+        return this.wizardService.state.context.selectedFacility;
+      },
+      facilityDescription() {
+        return this.formatNameAndId(this.facility.name, this.facility.id);
+      },
+      deviceDescription() {
+        if (this.device.name) {
+          return this.$tr('commaSeparatedPair', {
+            first: this.formatNameAndId(this.device.name, this.device.id),
+            second: this.device.baseurl,
+          });
+        }
+        return '';
+      },
+    },
+    beforeMount() {
+      this.isPolling = true;
+      this.pollImportTask();
+    },
+    methods: {
+      importedLearners() {
+        return this.wizardService.state.context.importedUsers;
+      },
+      pollImportTask() {
+        TaskResource.list({ queue: SoudQueue }).then(tasks => {
+          if (tasks.length) {
+            tasks.forEach(task => {
+              if (task.status === TaskStatuses.COMPLETED) {
+                // Remove completed user id from 'being imported'
+                const taskUserId = task.extra_metadata.user_id;
+                this.learnersBeingImported = this.learnersBeingImported.filter(
+                  id => id != taskUserId
+                );
+
+                // Update the wizard context to know this user has been imported - only if they
+                // haven't already been added to the list (ie, imported by other means)
+                const taskUsername = task.extra_metadata.username;
+                if (!this.importedLearners().includes(taskUsername)) {
+                  this.wizardService.send({
+                    type: 'ADD_IMPORTED_USER',
+                    value: taskUsername,
+                  });
+                }
+              }
+            });
+          }
+        });
+        if (this.isPolling) {
+          setTimeout(() => {
+            this.pollImportTask();
+          }, 2000);
+        }
```

Comment:
You might consider setting the next timeout after previous call has resolved. We have other places like this where we poll but don't wait for the previous call to finish, creating the chance for multiple concurrent requests if the server gets bogged down